avro/src/main/java/com/fasterxml/jackson/dataformat/avro/deser/ArrayReader.java

```diff
@@ -132,23 +133,24 @@ public JsonToken nextToken() throws IOException
 
     private final static class NonScalar extends ArrayReader
     {
-        private final AvroStructureReader _elementReader;
+        private AvroStructureReader _elementReader;
+        private final Supplier<AvroStructureReader> _elementReaderSupplier;
```

Comment:
For minimal refactoring, I used a supplier to defer the reading of the stream until after the Array has processed the required tokens